crates/forge-runner/src/running.rs

```diff
@@ -275,19 +279,58 @@ pub fn run_test_case(
         None
     };
 
-    let run_result = runner.run_function(
+    let mut vm = VirtualMachine::new(true);
+    let run_result = runner.run_function_with_vm(
         func,
+        &mut vm,
         &mut test_execution_syscall_handler,
         hints_dict,
         instructions,
         builtins,
     );
 
+    test_execution_syscall_handler
+        .child
+        .child
+        .child
+        .resources
+        .vm_resources += &test_execution_syscall_handler
+        .child
+        .child
+        .cheatnet_state
+        .used_resources
+        .vm_resources;
+    test_execution_syscall_handler
+        .child
+        .child
+        .child
+        .resources
+        .syscall_counter
+        .extend(
+            &test_execution_syscall_handler
+                .child
+                .child
+                .cheatnet_state
+                .used_resources
+                .syscall_counter,
+        );
```

Comment:
Could this at least be a separate function so I won't ever again have to look at it? 🙏